### DISCUSSION TITLE
Data-importer - Install error?

### DISCUSSION THREAD

**Discussion by Alpoy** [firefly-iii/firefly-iii#6931] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi,

I'm trying to install the data-importer (self-hosted) via composer and it always ends with an "error":
_Unable to set application key. No APP_KEY variable was found in the .env file._
As I understand the .env file comments/docs, there's not supposed to be a APP_KEY variable configured in the data-importer's .env file ? _"Do not use "APP_KEY". That's the WRONG one."_

I'm not sure if this is a problem at all, just I'm having trouble with some 500 - internal error messages and I can't say for sure my installation is correct because of this message at the end of install.

### Expected behaviour

A installation complete or other message verifying a correct install

### Debug information

> sh# composer create-project firefly-iii/data-importer --no-dev --prefer-dist data-importer 1.0.0
> Creating a "firefly-iii/data-importer" project at "./data-importer"
> Info from https://repo.packagist.org: #StandWithUkraine
> Installing firefly-iii/data-importer (v1.0.0)
>   - Installing firefly-iii/data-importer (v1.0.0): Extracting archive
> Created project in /var/www/data-importer
> > @php -r "file_exists('.env') || copy('.env.example', '.env');"
> Installing dependencies from lock file
> Verifying lock file contents can be installed on current platform.
> Package operations: 79 installs, 0 updates, 0 removals
>   - Installing psr/log (3.0.0): Extracting archive
>   - Installing psr/cache (3.0.0): Extracting archive
>   - Installing doctrine/event-manager (2.0.0): Extracting archive
>   - Installing doctrine/deprecations (v1.0.0): Extracting archive
>   - Installing doctrine/cache (2.2.0): Extracting archive
>   - Installing doctrine/dbal (3.5.3): Extracting archive
>   - Installing doctrine/inflector (2.0.6): Extracting archive
>   - Installing doctrine/lexer (3.0.0): Extracting archive
>   - Installing symfony/polyfill-ctype (v1.27.0): Extracting archive
>   - Installing webmozart/assert (1.11.0): Extracting archive
>   - Installing dragonmantank/cron-expression (v3.3.2): Extracting archive
>   - Installing voku/portable-ascii (2.0.1): Extracting archive
>   - Installing symfony/polyfill-php80 (v1.27.0): Extracting archive
>   - Installing symfony/polyfill-mbstring (v1.27.0): Extracting archive
>   - Installing phpoption/phpoption (1.9.0): Extracting archive
>   - Installing graham-campbell/result-type (v1.1.0): Extracting archive
>   - Installing vlucas/phpdotenv (v5.5.0): Extracting archive
>   - Installing symfony/css-selector (v6.2.3): Extracting archive
>   - Installing tijsverkoyen/css-to-inline-styles (2.2.6): Extracting archive
>   - Installing symfony/var-dumper (v6.2.3): Extracting archive
>   - Installing symfony/polyfill-uuid (v1.27.0): Extracting archive
>   - Installing symfony/uid (v6.2.0): Extracting archive
>   - Installing symfony/routing (v6.2.3): Extracting archive
>   - Installing symfony/process (v6.2.0): Extracting archive
>   - Installing symfony/polyfill-php72 (v1.27.0): Extracting archive
>   - Installing symfony/polyfill-intl-normalizer (v1.27.0): Extracting archive
>   - Installing symfony/polyfill-intl-idn (v1.27.0): Extracting archive
>   - Installing symfony/mime (v6.2.2): Extracting archive
>   - Installing psr/container (2.0.2): Extracting archive
>   - Installing symfony/service-contracts (v3.2.0): Extracting archive
>   - Installing psr/event-dispatcher (1.0.0): Extracting archive
>   - Installing symfony/event-dispatcher-contracts (v3.2.0): Extracting archive
>   - Installing symfony/event-dispatcher (v6.2.2): Extracting archive
>   - Installing egulias/email-validator (4.0.1): Extracting archive
>   - Installing symfony/mailer (6.3.x-dev 2e2ac75): Extracting archive
>   - Installing symfony/deprecation-contracts (v3.2.0): Extracting archive
>   - Installing symfony/http-foundation (v6.2.2): Extracting archive
>   - Installing symfony/error-handler (v6.2.3): Extracting archive
>   - Installing symfony/http-kernel (v6.2.4): Extracting archive
>   - Installing symfony/finder (v6.2.3): Extracting archive
>   - Installing symfony/polyfill-intl-grapheme (v1.27.0): Extracting archive
>   - Installing symfony/string (v6.2.2): Extracting archive
>   - Installing symfony/console (v6.2.3): Extracting archive
>   - Installing ramsey/collection (2.0.0): Extracting archive
>   - Installing brick/math (0.10.2): Extracting archive
>   - Installing ramsey/uuid (4.7.3): Extracting archive
>   - Installing psr/simple-cache (3.0.0): Extracting archive
>   - Installing nunomaduro/termwind (v1.15.0): Extracting archive
>   - Installing symfony/translation-contracts (v3.2.0): Extracting archive
>   - Installing symfony/translation (v6.2.3): Extracting archive
>   - Installing nesbot/carbon (2.65.0): Extracting archive
>   - Installing monolog/monolog (2.8.0): Extracting archive
>   - Installing league/mime-type-detection (1.11.0): Extracting archive
>   - Installing league/flysystem (3.12.2): Extracting archive
>   - Installing nette/utils (v3.2.9): Extracting archive
>   - Installing nette/schema (v1.2.3): Extracting archive
>   - Installing dflydev/dot-access-data (v3.0.2): Extracting archive
>   - Installing league/config (v1.2.0): Extracting archive
>   - Installing league/commonmark (2.3.8): Extracting archive
>   - Installing laravel/serializable-closure (v1.2.2): Extracting archive
>   - Installing fruitcake/php-cors (v1.2.0): Extracting archive
>   - Installing laravel/framework (v9.48.0): Extracting archive
>   - Installing fideloper/proxy (4.4.2): Extracting archive
>   - Installing psr/http-message (1.0.1): Extracting archive
>   - Installing psr/http-client (1.0.1): Extracting archive
>   - Installing ralouphie/getallheaders (3.0.3): Extracting archive
>   - Installing psr/http-factory (1.0.1): Extracting archive
>   - Installing guzzlehttp/psr7 (2.4.3): Extracting archive
>   - Installing guzzlehttp/promises (1.5.2): Extracting archive
>   - Installing guzzlehttp/guzzle (7.5.0): Extracting archive
>   - Installing firefly-iii/api-support-classes (dev-main 15a942e): Extracting archive
>   - Installing fruitcake/laravel-cors (v3.0.0): Extracting archive
>   - Installing nikic/php-parser (v4.15.3): Extracting archive
>   - Installing psy/psysh (v0.11.10): Extracting archive
>   - Installing laravel/tinker (v2.8.0): Extracting archive
>   - Installing league/csv (9.8.0): Extracting archive
>   - Installing twig/twig (v3.5.0): Extracting archive
>   - Installing rcrowe/twigbridge (v0.14.0): Extracting archive
>   - Installing spatie/enum (3.13.0): Extracting archive
> Generating optimized autoload files
> > Illuminate\Foundation\ComposerScripts::postAutoloadDump
> > @php artisan package:discover --ansi
> 
>    INFO  Discovering packages.
> 
>   fideloper/proxy ............................................................................................................................. DONE
>   fruitcake/laravel-cors ...................................................................................................................... DONE
>   laravel/tinker .............................................................................................................................. DONE
>   nesbot/carbon ............................................................................................................................... DONE
>   nunomaduro/termwind ......................................................................................................................... DONE
>   rcrowe/twigbridge ........................................................................................................................... DONE
> 
> 57 packages you are using are looking for funding.
> Use the `composer fund` command to find out more!
> No security vulnerability advisories found
> > @php artisan key:generate --ansi
> **Unable to set application key. No APP_KEY variable was found in the .env file.**
> sh#
### Steps to reproduce

Clean self-hosted install on debian 11 with php8.2

### Additional info

.

**Comment by JC5** [comment, marked as answer]
I see the error as well, but the installation proceeds as expected. I'll pick this up for 1.0.1, thanks!
